tests/FSharp.Compiler.ComponentTests/ErrorMessages/ClassesTests.fs

```diff
@@ -861,4 +861,83 @@ type C5 = class inherit System.MulticastDelegate override x.ToString() = ""  end
             (Error 771, Line 4, Col 25, Line 4, Col 36, "The types System.ValueType, System.Enum, System.Delegate, System.MulticastDelegate and System.Array cannot be used as super types in an object expression or class")
             (Error 771, Line 5, Col 25, Line 5, Col 40, "The types System.ValueType, System.Enum, System.Delegate, System.MulticastDelegate and System.Array cannot be used as super types in an object expression or class");
             (Error 771, Line 6, Col 25, Line 6, Col 49, "The types System.ValueType, System.Enum, System.Delegate, System.MulticastDelegate and System.Array cannot be used as super types in an object expression or class")
+        ]
+        
+
+    [<Fact>]
+    let ``Types can inherit from a single concrete type`` () =
+        Fsx """
+type ClassA() = class end
+
+type Class() =
+    inherit ClassA()
+        """
+        |> typecheck
+        |> shouldSucceed
+
+    [<Fact>]
+    let ``Types cannot inherit from multiple concrete types.`` () =
+        Fsx """
+type ClassA() = class end
+
+type ClassB() = class end
+
+type ClassC() = class end
+
+type Class() =
+    inherit ClassA()
+    inherit ClassB()
+    inherit ClassC()
+        """
+        |> typecheck
+        |> shouldFail
+        |> withDiagnostics [
+            (Error 959, Line 8, Col 6, Line 8, Col 11, "Type definitions may only have one 'inherit' specification and it must be the first declaration")
+            (Error 932, Line 10, Col 13, Line 10, Col 19, "Types cannot inherit from multiple concrete types")
+            (Error 932, Line 11, Col 13, Line 11, Col 19, "Types cannot inherit from multiple concrete types")
+        ]
+        
+    [<Fact>]
+    let ``Types cannot inherit from multiple concrete types. Type name cannot be empty.`` () =
+        Fsx """
+type IA = interface end
+
+type I =
+    inherit IA
+    inherit
+        """
+        |> typecheck
+        |> shouldFail
+        |> withDiagnostics [
+            (Error 3159, Line 6, Col 5, Line 6, Col 12, "Type name cannot be empty.")
+        ]
+
+    [<Fact>]
+    let ``Inheriting multiple base interfaces`` () =
+        Fsx """
+type IA = interface end
+type IB = interface end
+
+type I =
+    inherit IA
+    inherit IB
+        """
+        |> typecheck
+        |> shouldSucceed
+    
+    [<Fact>]
+    let ``Class inheriting multiple base interfaces`` () =
+        Fsx """
+type IA = interface end
+type IB = interface end
+
+type I() =
+    inherit IA
+    inherit IB
+        """
+        |> typecheck
+        |> shouldFail
+        |> withDiagnostics [
+            (Error 961, Line 6, Col 5, Line 6, Col 12, "This 'inherit' declaration specifies the inherited type but no arguments. Consider supplying arguments, e.g. 'inherit BaseType(args)'.")
```

Comment:
This is a bit misleading here... but looks like it's been like this before, so just pointing out.

Reply:
@psfinaki 

Yes it has been like this. What do you think we should do here ? . The main issue here is that you can not from an interface when using a class, only with a type.

![Screenshot 2024-10-25 at 15 43 57](https://github.com/user-attachments/assets/d00e20f3-ac88-4563-b0c6-1bc772c05c37)

I think we should replace  `961` with  `error FS0946: Cannot inherit from interface type. Use interface ... with instead`

Reply:
Yeah IMO the first error is just redundant. We show two errors now, I think we could leave just 946 here.